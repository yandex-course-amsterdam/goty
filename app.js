const { app } = require('./build/app')

const PORT = process.env.PORT || 5000

app.listen(process.env.PORT || PORT)
