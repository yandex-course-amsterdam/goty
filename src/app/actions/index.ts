export * from './types'
export * from './userInfo'
