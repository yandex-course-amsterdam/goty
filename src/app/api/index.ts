export { AuthApi } from './AuthApi'
export { Api } from './Api'
export { UsersApi } from './UsersApi'
