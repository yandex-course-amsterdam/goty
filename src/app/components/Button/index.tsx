import React, { ReactElement } from 'react'
import cn from 'classnames'

import style from './style.css'

type ButtonProps = {
  type: 'button' | 'submit'
  buttonText: string
  className?: string
  handleClick?: () => void
}

export const Button = ({ type, buttonText, className, handleClick }: ButtonProps): ReactElement => {
  return (
    <button
      onClick={handleClick}
      className={cn(style.button, className)}
      type={type === 'submit' ? 'submit' : 'button'}
      disabled
    >
      {buttonText}
    </button>
  )
}
