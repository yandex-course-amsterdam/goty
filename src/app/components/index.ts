export { Sidebar } from './Sidebar'
export { Avatar } from './Avatar'
export { Navigation } from './Navigation'
export { Main } from './Main'
export { Title } from './Title'
export { Description } from './Description'
export { Input } from './Input'
export { Textarea } from './Textarea'
export { Label } from './Label'
export { Error } from './Error'
export { ErrorBoundary } from './ErrorBoundary'
export { OfflineBar } from './OfflineBar'
export { Button } from './Button'
export { SubNavigation } from './SubNavigation'
export { Canvas } from './Canvas'
export { UserScore } from './UserScore'
export { Comment } from './Comment'
export { Comments } from './Comments'
export { Likes } from './Likes'
export { NewsList } from './NewsList'
export { SignUpForm } from './Forms/SignUpForm'
export { SignInForm } from './Forms/SignInForm'
export { FeedbackForm } from './Forms/FeedbackForm'
export { DetailsForm } from './Forms/DetailsForm'
export { PasswordForm } from './Forms/PasswordForm'
export { AvatarForm } from './Forms/AvatarForm'
export { Loader } from './Loader'
export { PrivateRoute } from './PrivateRoute'
export { ExitButton } from './ExitButton'
export { Authorization } from './Authorization'
export { Theme } from './Theme'
export { App } from './App'
