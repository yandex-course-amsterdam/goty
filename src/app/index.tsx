export { App as Game } from 'app/containers/App'
