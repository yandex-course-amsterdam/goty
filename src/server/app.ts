import path from 'path'

import express from 'express'
import compression from 'compression'
import cookieParser from 'cookie-parser'

import {
  authRouter,
  usersRouter,
  themesRouter,
  feedbackRouter,
  newsRouter,
  commentsRouter,
  likeRouter
} from './routes'

import { serverRenderMiddleware } from './serverRenderMiddleware'

const app = express()

app
  .use(express.json())
  .use(cookieParser())
  .use(compression())
  .use('/', express.static(__dirname))
  .use('/auth', authRouter)
  .use('/users', usersRouter)
  .use('/themes', themesRouter)
  .use('/feedback', feedbackRouter)
  .use('/news', newsRouter)
  .use('/comments', commentsRouter)
  .use('/likes', likeRouter)

app.get('/images/:src', (req, res) => {
  const { src } = req.params
  const options = {
    root: path.join(__dirname, '../')
  }

  // TODO: перенести статику в build/
  res.sendFile(`/src/images/${src}`, options)
})

app.get('*', serverRenderMiddleware)

export { app }
