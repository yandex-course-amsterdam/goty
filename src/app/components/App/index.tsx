import React, { FC, useCallback, useEffect, useState } from 'react'
import { useSelector, useDispatch } from 'react-redux'

import { Switch, Route } from 'react-router-dom'
import { CSSTransition } from 'react-transition-group'

import {
  SignUpView,
  SignInView,
  ProfileView,
  GameView,
  ScoreView,
  NotFoundView
} from 'app/views'
import { OfflineBar, PrivateRoute, Authorization } from 'app/components'

import { postResult, getUserTheme } from 'app/api/Api'
import { fetchUserInfo } from 'app/actions'

import { StoreState } from 'app/reducers'
import { route } from 'app/enums'
import {
  getScore,
  removeScore,
  isServer,
  setUserTheme,
  storeUserTheme
} from 'app/utils'

import 'normalize.css'
import 'assets/main.css'
import '../../../fonts/fonts.css'

// startServiceWorker()

export const App: FC = (): JSX.Element => {
  const [isOffline, setIsOffline] = useState(isServer || !navigator.onLine)

  const dispatch = useDispatch()

  const loginStatus = useSelector(
    (state: StoreState) => state.loginStatus.status
  )

  const handleOfflineCallback = useCallback(() => setIsOffline(true), [])
  const handleOnlineCallback = useCallback(() => {
    setIsOffline(false)

    const storedScore = getScore()

    if (storedScore) {
      postResult(storedScore)
      removeScore()
    }
  }, [])

  useEffect(() => {
    const fetchUser = async () => {
      try {
        const data = await dispatch(fetchUserInfo())
        // Очередной странный тайпинг, учитывая возвращаемое по факту значение
        // Сейчас уже нет времени разбираться, буду потом сильнее вникать в ts
        // @ts-ignore
        const { data: theme } = await getUserTheme(data.id)
        setUserTheme(theme)
        storeUserTheme(theme)
      } catch (error) {
        console.warn(error)
      }
    }

    if (loginStatus) {
      fetchUser()
    }

    if (!isServer) {
      window.addEventListener('offline', handleOfflineCallback)
      window.addEventListener('online', handleOnlineCallback)
    }

    return () => {
      if (!isServer) {
        window.removeEventListener('offline', handleOfflineCallback)
        window.removeEventListener('online', handleOnlineCallback)
      }
    }
  }, [])

  return (
    <>
      <Switch>
        <Route exact path={route.auth}>
          <Authorization />
        </Route>
        <Route path={route.signUp}>
          <SignUpView />
        </Route>
        <Route path={route.signIn}>
          <SignInView />
        </Route>
        <PrivateRoute path={route.profile}>
          <ProfileView />
        </PrivateRoute>
        <PrivateRoute path={route.game}>
          <GameView />
        </PrivateRoute>
        <PrivateRoute path={route.score}>
          <ScoreView />
        </PrivateRoute>
        <Route path="*">
          <NotFoundView />
        </Route>
      </Switch>

      <CSSTransition
        in={isOffline}
        appear={isOffline}
        timeout={0}
        classNames="bar"
      >
        <OfflineBar />
      </CSSTransition>
    </>
  )
}
