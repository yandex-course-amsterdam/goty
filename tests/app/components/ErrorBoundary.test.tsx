import React from 'react'
import * as renderer from 'react-test-renderer'

import { ErrorBoundary } from 'app/components/'
import { Button } from 'app/components/Button'
import { Throwable } from './mockComponents/Throwable'

describe('Error component', () => {
  test('Render children when there are no errors', () => {
    const component = renderer
      .create(
        <ErrorBoundary>
          <Button buttonText="Oh hi" />
        </ErrorBoundary>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })

  test('Render fallback UI when there is an error', () => {
    const component = renderer
      .create(
        <ErrorBoundary>
          <Throwable />
        </ErrorBoundary>
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
