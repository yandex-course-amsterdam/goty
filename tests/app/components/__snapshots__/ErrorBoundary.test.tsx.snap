// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Boundary component Render children when there are no errors 1`] = `
<button
  className=""
  disabled={false}
  type="button"
>
  Oh hi
</button>
`;

exports[`Error Boundary component Render fallback UI when there is an error 1`] = `
<div>
  <h2
    className=""
  >
    Oops
  </h2>
  <div>
    <div
      className=""
    >
      <p>
        Something went wrong
      </p>
      <p>
        Try to refresh page or visit another one.
      </p>
    </div>
  </div>
</div>
`;
